[opencv] update to 4.9

cenit opened the pull request:
Fixes #41418
Fixes #41250 
Fixes #41094 
Fixes #41038 
Fixes #40680 
Fixes #40568 
Fixes #40393 
Fixes #40406 
Fixes #39884
Fixes #39835
Fixes #39651
Fixes #39454 
Fixes #39224
Fixes #38321 
Fixes #36771 
Fixes #36093
Fixes #36009
Fixes #35937 
Fixes #34559 
Fixes #34279 
Fixes #33742 
Fixes #32228
Fixes #17850 
Fixes #13120 
Fixes #4937 

~Depending on #39703 (need to remove ffmpeg from ci.baseline here after that PR acceptance)~

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [x] The "supports" clause reflects platforms that may be fixed by this new version.
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

cenit committed:
scaffolding 4.9 update

cenit committed:
update patches for base opencv

cenit committed:
[opencv4] fix residual issues with basic build

cenit committed:
[fix manifest

cenit committed:
fix references

cenit committed:
fix referencing ports

cenit committed:
fix references

cenit committed:
fix additional download dependencies

cenit committed:
fix references

cenit committed:
fix call

cenit committed:
fix references

Neumann-A commented on ports/opencv4/portfile.cmake:
```diff
@@ -56,23 +49,31 @@ set(ADE_DIR ${CURRENT_INSTALLED_DIR}/share/ade CACHE PATH "Path to existing ADE
  "jpeg"      WITH_JPEG
  "lapack"    WITH_LAPACK
  "nonfree"   OPENCV_ENABLE_NONFREE
+ "nofs"      OPENCV_DISABLE_FILESYSTEM_SUPPORT
+ "nothread"  OPENCV_DISABLE_THREAD_SUPPORT
```
Features should be positive not negative. Reason: Features are meant to be additive not subtractive. It also makes the feature name shorter.

cenit replied:
should be done

Neumann-A commented on ports/opencv4/portfile.cmake:
```diff
@@ -564,7 +584,7 @@ find_dependency(Tesseract)")
 set(CMAKE_AUTORCC ON)
 set(CMAKE_AUTOUIC ON)
-find_dependency(Qt${USE_QT_VERSION} COMPONENTS Core Gui Widgets Test Concurrent Core5Compat)")
+find_dependency(Qt${USE_QT_VERSION} COMPONENTS Core Gui Widgets Test Concurrent ${QT_CORE5COMPAT})")
```
what does this do in regards to vcpkg?

cenit replied:
at the beginning of the portfile there is the easy selector for qt version dependency (in case of overlays, i assume). But the dependency for Core5Compat was brought in even when QT5 was selected

Neumann-A commented on ports/opencv4/portfile.cmake:
```diff
@@ -471,9 +484,16 @@ vcpkg_cmake_configure(
+        -DWITH_DSHOW=${WITH_DSHOW}
+        -DBUILD_opencv_calib3d=${BUILD_opencv_calib3d}
+        -DBUILD_opencv_highgui=${BUILD_opencv_highgui}
+        ${ADDITIONAL_BUILD_FLAGS}
```
Please also fix python installation behavior.
I needed to add:
`        -D__INSTALL_PATH_PYTHON3=${CURRENT_PACKAGES_DIR}/${PYTHON3_SITE}/cv2`

and for cleanup later:

```
  file(GLOB python_dir LIST_DIRECTORIES true RELATIVE "${CURRENT_PACKAGES_DIR}/lib/" "${CURRENT_PACKAGES_DIR}/lib/python*")
  file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/lib/${python_dir}/site-packages/cv2/typing")
  file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/lib/${python_dir}/site-packages/cv2/typing")
```

However if the install logs show that python bindings get installed into the right directory already than consider this obsolete.

cenit replied:
i am fixing this point then i hope the work on the PR is finished

Neumann-A commented on ports/opencv4/portfile.cmake:
```diff
@@ -471,9 +484,16 @@ vcpkg_cmake_configure(
-        ${ADDITIONAL_BUILD_FLAGS}
         -DBUILD_IPP_IW=${WITH_IPP}
         -DOPENCV_LAPACK_FIND_PACKAGE_ONLY=ON
+        -DOPENCV_DISABLE_FILESYSTEM_SUPPORT=${OPENCV_DISABLE_FILESYSTEM_SUPPORT}
+        -DWITH_OPENCL=${WITH_OPENCL}
+        -DWITH_WIN32UI=${WITH_WIN32UI}
+        -DCV_ENABLE_INTRINSICS=${CV_ENABLE_INTRINSICS}
+        -DWITH_DSHOW=${WITH_DSHOW}
+        -DBUILD_opencv_calib3d=${BUILD_opencv_calib3d}
+        -DBUILD_opencv_highgui=${BUILD_opencv_highgui}
```
Shouldn't this be controlled by `vcpkg_check_features`?

Neumann-A commented on ports/opencv4/portfile.cmake:
```diff
@@ -380,12 +383,16 @@ endif()
 vcpkg_cmake_configure(
```
I feel like there are way to many options here which should and are already be controlled by `vcpkg_check_features`

```
        -DBUILD_WEBP=OFF
        -DBUILD_ZLIB=OFF
        -DBUILD_TBB=OFF
        -DWITH_OPENVINO=${WITH_OPENVINO}
        -DWITH_TBB=${WITH_TBB}
        -DWITH_OPENJPEG=OFF

```
etc.

cenit replied:
some *with* options are more complex than what might be handled by vcpkg_check_features and so they are handled outside of it

cenit replied:
the combination to decide on/off is not just user based but depends on the target architecture/os...

Neumann-A replied:
> the combination to decide on/off is not just user based but depends on the target architecture/os...

That is controlled by designing the features accordingly with support statements or making them default features + platform expression. In general you don't need to have os/arch based logic if `vcpkg_check_features`  is enough to pass the flag. The ON/OFF selection can be done via designing the manifest accordingly.

cenit replied:
ok I am leaving all your suggestions as open comments to remind them. Right now my first goal is to have it green flagged in ci, then I will do these "cosmetic" fixes

Neumann-A commented on ports/opencv4/portfile.cmake:
```diff
@@ -47,6 +36,10 @@ else()
   set(TARGET_IS_X86 1)
 endif()
 
+if (USE_QT_VERSION STREQUAL "6")
+  set(QT_CORE5COMPAT "Core5Compat")
+endif()
+
```
Qt version in vcpkg should always be 6

cenit replied:
there is a switch at the top of the portfile, this is just to add consistency in case in an overlay you put "5"
it's very convenient to write more easily overlays, i am not sure the removal would be ok for everybody

cenit committed:
fix for cuda first class language, enabled only if cuda is enabled

cenit committed:
Merge remote-tracking branch 'upstream/master' into dev/cenit/opencv49

cenit committed:
Merge branch 'dev/cenit/opencv49' of https://github.com/cenit/vcpkg into dev/cenit/opencv49

cenit committed:
disable some problematic features to check the others

cenit committed:
fix include folder

cenit committed:
fix older versions

cenit committed:
fix references

cenit committed:
remove missing features also from depending ports

cenit committed:
fix references

cenit committed:
Merge branch 'master' into dev/cenit/opencv49

cenit committed:
Merge branch 'master' into dev/cenit/opencv49

cenit committed:
update references

cenit committed:
remove missing features

cenit committed:
update references

cenit committed:
fix sha

cenit committed:
fix reference

cenit committed:
remove temporarily lapack feature from testing

cenit committed:
update reference

cenit committed:
add even more patches for cuda 12.4+ compatibility

cenit committed:
fix reference

cenit committed:
add some more patches for cuda 12.4+ compatibility

cenit committed:
fix reference

cenit committed:
add arm64 patch

cenit committed:
update reference

cenit commented:
i want to close here quickly because then there is already opencv 4.10 to be added!

cenit committed:
use c++17

cenit committed:
Merge branch 'master' into dev/cenit/opencv49

cenit committed:
moving some conditions to manifest
update feature matrix
lapack was enabled without notice without this further variable
remove cmake target
win32ui not available for uwp
add fixme tags
update reference

cenit committed:
moving some conditions to manifest,
update feature matrix,
lapack was enabled without notice without this further variable,
remove cmake target,
win32ui not available for uwp,
add fixme tags,
remove gtk feature from default,
update reference

cenit committed:
Merge branch 'dev/cenit/opencv49' of https://github.com/cenit/vcpkg into dev/cenit/opencv49

cenit commented:
opencv and opencv4 are marked as passing on the remaining "failing tests" but vcpkg-ci-opencv is marked as failing without even testing. What am i missing?

cenit committed:
Merge branch 'master' into dev/cenit/opencv49

cenit committed:
Merge branch 'dev/cenit/opencv49' of https://github.com/cenit/vcpkg into dev/cenit/opencv49

cenit committed:
move opencv2 and opencv3 clauses all to manifests

cenit committed:
format manifest

cenit committed:
format manifest

cenit committed:
update references

cenit committed:
improve vcpkg-ci-opencv to cover all opencv versions

cenit committed:
fix reference

cenit commented:
i ported opencv2 and opencv3 features to manifest and improved their testing policies
i will port opencv4 features to manifest asap

cenit committed:
add opencv3.pc patch

cenit committed:
fix reference

cenit committed:
fix patch name

cenit committed:
fix reference

cenit committed:
move opencv4 features to manifest

cenit committed:
fix manifest

cenit committed:
fix references

cenit committed:
add more pkgconf manual fixes

cenit committed:
fix references

cenit committed:
add more pkgconf manual fixes - part 2

cenit committed:
remove broken features from ci

cenit committed:
fix references

cenit commented:
ok now all features should have been moved to manifest. Please let me know

cenit committed:
fix reference

cenit committed:
fix reference

cenit committed:
fix ffmpeg, don't know where cascade is coming from, locally it works

cenit commented:
locally it is building fine, but on azure i see sometimes the vcpkg-ci-port marked as cascade. Cannot reproduce it locally, can you help me disentangle the source of it?

cenit committed:
exclude android from opencv[python] ci coverage

cenit committed:
restore deleted version from registry

cenit committed:
remove ffmpeg features from opencv ci coverage on android

cenit committed:
fix feature matrix

cenit committed:
Merge branch 'master' into dev/cenit/opencv49

cenit committed:
bump port version

cenit commented:
we stopped short of testing halide feature in opencv because it was including llvm, but somehow CI is still building it anyway for another port maybe... and i assume it's not able to store it in the blob storage. This bug is still with us years after 😄

cenit committed:
fix metaport

cenit committed:
fix reference

cenit committed:
add missing version define

cenit committed:
fix reference

cenit committed:
[openvino] add pkgconfig

cenit committed:
add references

dg0yt commented:
Yeah, the artifact caching problems (llvm et al.) become more painful. IIUC we could make the llvm tools use a shared lib even in static builds. But that doesn't belong to this PR ;-)

dg0yt reviewed (commented):
Touching ci.baseline.txt means building what isn't cached, even if unchanged. That's how we trigger llvm with every port improvement.

cenit commented on scripts/ci.baseline.txt:
```diff
@@ -330,6 +330,9 @@ fastrtps:arm64-android=fail
 fastrtps:x64-android=fail
 fastrtps:x64-windows-static-md=fail
 fastrtps:x64-windows-static=fail
+ffmpeg:arm-neon-android=fail
+ffmpeg:arm64-android=fail
+ffmpeg:x64-android=fail
```
i know, but it was not even tested before... i unlocked it in ci but i cannot really fix all ports here...

cenit replied:
(was untested due to unsupported combination of features in "support" clauses in vcpkg-ci-ffmpeg)

cenit replied:
(also, i don't even have an android toolchain installed to verify what's broken)

cenit commented on scripts/ci.baseline.txt:
```diff
@@ -771,6 +771,9 @@ moos-essential:x64-windows-static-md=fail
 moos-essential:x64-windows=fail
 moos-essential:x86-windows=fail
 # Conflicts with libjpeg-turbo
+mozjpeg:arm-neon-android=fail
+mozjpeg:arm64-android=fail
+mozjpeg:x64-android=fail
 mozjpeg:arm64-windows      = skip
```
don't know, just moved it next to others while i was looking for it

dg0yt replied:
I suggest to not move it, or to rectify as needed.

cenit commented:
now there is a problem with openvino on master. Everything was ok before the latest update that happened with it on master

cenit committed:
Merge branch 'master' into dev/cenit/opencv49

cenit committed:
add some more features

cenit committed:
fix references

cenit committed:
fix bugs in features handling

cenit committed:
fix references

cenit committed:
fix hdf5 wrong dependency

cenit committed:
fix reference

cenit committed:
Merge branch 'master' into dev/cenit/opencv49

cenit committed:
fix reference

cenit committed:
blind test to remove broken feature

cenit committed:
fix reference

dg0yt commented:
I guess the opencv2 problem with gtk is that it actually wants what is now gtk3 in vcpkg.

cenit commented:
i removed gtk altogether but the problem is persisting

cenit committed:
trace expand

cenit committed:
Merge branch 'master' into dev/cenit/opencv49

cenit committed:
remove gtk

howard0su committed:
[soapysdr] New Port (#39729)

nlogozzo committed:
[libnick] Update to 2024.7.2 (#39790)

sidy3d committed:
[nativefiledialog-extended] Update to version 1.2.0 (#39783)

c8ef committed:
[folly related] update to 07.08 (#39767)

autoantwort committed:
[v8] use right var (#39779)

autoantwort committed:
[liblsquic] use right var (#39781)

autoantwort committed:
[liblas] use right var (#39780)

autoantwort committed:
[libsrt] use right var (#39777)

autoantwort committed:
[nanopb] use right var (#39776)

autoantwort committed:
[libsvm] use right var (#39775)

autoantwort committed:
[imgui] fix definitions (#39774)

autoantwort committed:
[zlmediakit] fix core build (#39773)

autoantwort committed:
[ableton-link] use right variables (#39769)

jimwang118 committed:
[mnn] Fix feature opencl compilation error under Linux (#39794)

jimwang118 committed:
[crashpad] Fix build errors under linux (#39671)

LilyWangLL committed:
[fftw3] Fix the wrong package version (#39753)

Co-authored-by: Lily Wang <v-lilywang@microsoft.com>

m-kuhn committed:
[dbus] does not build on android and ios (#39750)

maharmstone committed:
[tdscpp] Add new port (#39599)

dg0yt committed:
[libxml2] Update to 2.11.8 [CVE] (#39704)

dg0yt committed:
[Skia] Update to chrome/127 (#39462)

Co-authored-by: Alexandre Bique <bique.alexandre@gmail.com>

dg0yt committed:
[sdl2] Fix exported deps for android (#39709)

kp-cat committed:
[configcat] Update to version 4.0.3 (#39714)

JakubMelka committed:
[blend2d] update port to 2024-06-28 (#39730)

Co-authored-by: Kai Pastor <dg0yt@darc.de>

bansan85 committed:
[itk] update to 5.4.0 (#39757)

harris-josh committed:
[thrift] Update thrift 0.20.0 git reference (#39787)

teo-tsirpanis committed:
[google-cloud-cpp] Skip building mock libraries. (#39802)

SHIINASAMA committed:
[sese] fix fmt error (#39801)

JavierMatosD committed:
[CI][GTK] Fix vulkan_enabled bug in GTK (#39788)

Co-authored-by: Javier Matos <javiermatos@Javiers-Laptop.local>

Pravila00 committed:
[opentelemetry-cpp] Fix OTLP build (#39823)

jimwang118 committed:
[vcpkg baseline][cnats] Passing remove from fail list (#39817)

JacobBarthelmeh committed:
[wolfssl] update to 5.7.2 (#39814)

autoantwort committed:
[lz4] update license expression (#39813)

Co-authored-by: Kai Pastor <dg0yt@darc.de>

myd7349 committed:
[yyjson] Update to 0.10.0 (#39827)

jimwang118 committed:
[librdkafka] Fix code download URL (#39841)

BillyONeal committed:
Attempt to disable CodeQL (#39844)

FrankXie05 committed:
[paho-mqttpp3] Update version to 1.4.1 (#39843)

Cheney-W committed:
[fastio] update to 2024-07-05 (#39845)

autoantwort committed:
[ctbignum] fix build (#39851)

wangfangkai committed:
[hdf5] fix: add support for aarch64 _Float16 16-bit floating point type (#39798)

Co-authored-by: 王方凯 <fk.wang@hydro-tech.cn>

walbourn committed:
[toolchain] Fix try compile handling for Xbox (#39654)

JAicewizard committed:
Don't unset CMAKE_CROSSCOMPILING when cross-compiling to linux (#37734)

cenit committed:
[opencv2] fix multiconfig generator

cenit committed:
Merge branch 'master' into dev/cenit/opencv49

cenit commented:
the residual error on openvino due to pkgconf missing is triggered by a bug in cmake.
The error is
`CMake Error at FindOpenSSL.cmake:207 (pkg_check_modules): Unknown CMake command "pkg_check_modules".`

This is in fact the snippet in FindOpenSSL.cmake official module:

```
if (UNIX)
  find_package(PkgConfig QUIET)
  pkg_check_modules(_OPENSSL QUIET openssl)
endif ()
```

as you can see, the module is using `pkg_check_modules` after a find_package "QUIET". In case the find fails, you don't have the command available and it should fail more gracefully...
I tried adding some lines to openvino port to make it sure that pkgconf is there, but without success. I am not sure this PR is responsible to make openvino work... i don't know why it's not failing on master

cenit committed:
Merge branch 'master' into dev/cenit/opencv49

cenit committed:
Merge branch 'master' into dev/cenit/opencv49

cenit committed:
re-enable ffmpeg on android

dg0yt commented:
> the module is using `pkg_check_modules` after a find_package "QUIET". In case the find fails, you don't have the command available and it should fail more gracefully...

AFAICT `pkg_check_modules` is always defined after `find_package(PkgConfig)`. CMake always loads the whole module. The macro just won't be successful if there is no `pkg-config` executable.

cenit committed:
select opencv version in a more reliable way

cenit committed:
update reference

cenit committed:
Merge branch 'master' into dev/cenit/opencv49

cenit committed:
fix usages

cenit committed:
fix references

cenit committed:
Merge branch 'dev/cenit/opencv49' of https://github.com/cenit/vcpkg into dev/cenit/opencv49

lwqwag commented:
any update?

cenit committed:
Merge branch 'master' into dev/cenit/opencv49

dg0yt commented:
openvino stumbles over FindOpenSSL.cmake not getting `pkg_check_modules` -  something messing with CMake module path?

cenit committed:
Merge branch 'master' into dev/cenit/opencv49

cenit committed:
Merge branch 'master' into dev/cenit/opencv49

cenit committed:
bump port version

cenit committed:
fix reference

dg0yt commented on scripts/ci.baseline.txt:
```diff
@@ -840,27 +840,6 @@ omplapp:x64-linux=skip
 onednn:x64-android=fail
 # opencc/deps/marisa-0.2.5/lib/marisa/grimoire/io/mapper.cc currently doesn't support UWP.
 opencc:x64-android=fail
-opencv2:arm-neon-android=fail
-opencv2:arm64-android=fail
-opencv2:arm64-windows         = skip
-opencv2:arm64-uwp               = skip
-opencv2:x64-android=fail
-opencv2:x64-linux             = skip
-opencv2:x64-osx               = skip
-opencv2:x64-uwp               = skip
-opencv2:x64-windows           = skip
-opencv2:x64-windows-static    = skip
-opencv2:x64-windows-static-md = skip
-opencv2:x86-windows           = skip
-opencv3:arm64-windows         = skip
-opencv3:arm64-uwp               = skip
-opencv3:x64-linux             = skip
-opencv3:x64-osx               = skip
-opencv3:x64-uwp               = skip
-opencv3:x64-windows           = skip
-opencv3:x64-windows-static    = skip
-opencv3:x64-windows-static-md = skip
-opencv3:x86-windows           = skip
```
AFAIU these entries were added to avoid problems in CI.
I can't see direct file conflicts. I assume that the problem is lack of control which major version of opencv is picked in (some) reverse dependencies, eventually leading to installation order problems and baseline regressions.

cenit replied:
technically that's the goal of the opencv port. It hides all other ports if not explicitly requested.

dg0yt replied:
Okay, I see what it is doing since #25431. But why do we carry these baseline entries then? Does every reverse dependency handle opencv lookup in that particular way, `find_package(OpenCV CONFIG)`?

cenit committed:
Merge branch 'master' into dev/cenit/opencv49

cenit committed:
bump ocv3 port version

cenit committed:
fix references

cenit committed:
Merge branch 'master' into dev/cenit/opencv49

cenit committed:
Merge branch 'master' into dev/cenit/opencv49

cenit committed:
Merge branch 'master' into dev/cenit/opencv49

cenit committed:
Merge branch 'master' into dev/cenit/opencv49

cenit committed:
[openvino] remove unnecessary pkgconfig retrieval

cenit committed:
add missing downstream find_dependency

cenit committed:
fix references

cenit committed:
Merge branch 'master' into dev/cenit/opencv49

cenit committed:
reference proper ports in vcpkg-ci-opencv, otherwise wrapper is not installed

cenit committed:
better uniformity between opencv3 and opencv4

cenit committed:
fix references

cenit committed:
fix inconsistencies

cenit committed:
fix references

cenit committed:
Merge branch 'master' into dev/cenit/opencv49

dg0yt commented on ports/openimageio/fix-dependencies.patch:
```diff
@@ -12,6 +12,9 @@ diff -u -r a/src/cmake/Config.cmake.in b/src/cmake/Config.cmake.in
 +if(@USE_LIBHEIF@)
 +    find_dependency(libheif CONFIG)
 +endif()
++if(@USE_FREETYPE@)
++    find_dependency(freetype CONFIG)
++endif()
```
We should move this to a separate PR if this PR takes more time.

cenit committed:
fix missing clause

cenit committed:
fix references

cenit committed:
[pcl] fix regression

cenit committed:
fix patch compatibility with opencv 3

cenit committed:
install to packages to not break cache

cenit committed:
fix references

cenit commented:
check is green

dg0yt commented on ports/opencv2/usage.in:
```diff
@@ -1,4 +1,4 @@
-The package @PORT@ is compatible with CMake if you set the OpenCV_DIR *before* the find_package call
+In case of multiple OpenCV version installed, set the OpenCV_ROOT *before* the find_package call
```
```suggestion
@PORT@ is compatible with built-in CMake variables. In case of multiple OpenCV version installed, set OpenCV_ROOT *before* the find_package call:
```

dg0yt commented on ports/opencv3/0015-fix-supportqnx.patch:
```diff
@@ -1,5 +1,3 @@
-diff --git a/modules/core/src/system.cpp b/modules/core/src/system.cpp
-index 9bade08..28f006e 100644
```
I assume that removing these lines make git treat this patch as a new file. It isn't really necessary. (Same for some patches.)

(Which git option should we use to avoid these lines for new patches?)

cenit replied:
I remove all the lines that are unnecessary and boiler plate from patch files, for clarity and better recyclability. This is a standard procedure i am trying to always adopt here on vcpkg, because that's what minimizes differences in the long run, if you never write them down

dg0yt replied:
Well, I'm all for minimal changes, but I wonder if this isn't adding more steps to future workflows if it can't be handled with a simple git option.

dg0yt commented on ports/opencv4/usage.in:
```diff
@@ -1,5 +1,4 @@
-If you do not install the meta-port *opencv*, the package @PORT@ is compatible with CMake
-if you set the OpenCV_DIR *before* the find_package call
+In case of multiple OpenCV version installed, set the OpenCV_ROOT *before* the find_package call
```
```suggestion
@PORT@ is compatible with built-in CMake variables. In case of multiple OpenCV version installed, set OpenCV_ROOT *before* the find_package call:
```

dg0yt commented on ports/opencv3/usage.in:
```diff
@@ -1,4 +1,4 @@
-The package @PORT@ is compatible with CMake if you set the OpenCV_DIR *before* the find_package call
+In case of multiple OpenCV version installed, set the OpenCV_ROOT *before* the find_package call
```
```suggestion
@PORT@ is compatible with built-in CMake variables. In case of multiple OpenCV version installed, set OpenCV_ROOT *before* the find_package call:
```

dg0yt commented on ports/opencv4/0015-fix-freetype.patch:
```diff
@@ -1,6 +1,6 @@
 --- a/modules/gapi/cmake/init.cmake
 +++ b/modules/gapi/cmake/init.cmake
-@@ -20,7 +20,7 @@ if(NOT TARGET ade)
+@@ -21,7 +21,7 @@ if(NOT TARGET ade)
```
Such patches could go unchanged.

cenit replied:
since i am regenerating them anyway, better to make them work properly also for people having strict requirements on patch validation

dg0yt commented on ports/opencv4/vcpkg.json:
```diff
@@ -125,6 +157,9 @@
       ]
     },
+    "dshow": {
+      "description": "enable DirectShow"
```
```suggestion
      "description": "Enable DirectShow"
```

dg0yt commented on ports/opencv4/vcpkg.json:
```diff
@@ -203,8 +245,15 @@
+      "description": "highgui module"
+    },
+    "intrinsics": {
+      "description": "enable intrinsics"
```
```suggestion
      "description": "Enable intrinsics"
```

dg0yt commented on ports/opencv4/vcpkg.json:
```diff
@@ -225,9 +274,19 @@
       "description": "allow nonfree and unredistributable libraries"
     },
+    "opencl": {
+      "description": "enable opencl support",
```
```suggestion
      "description": "Enable opencl support",
```

dg0yt commented on ports/opencv4/vcpkg.json:
```diff
@@ -247,7 +306,8 @@
     },
     "openmp": {
-      "description": "Enable openmp support for opencv"
+      "description": "Enable openmp support for opencv",
```
```suggestion
      "description": "Enable OpenMP support",
```

dg0yt commented on ports/opencv4/vcpkg.json:
```diff
@@ -396,6 +487,9 @@
       ]
     },
+    "win32ui": {
+      "description": "enable win32ui"
```
```suggestion
      "description": "Enable win32ui"
```
I don't know what is but I would guess it is limited to `windows`.

cenit replied:
windows & !uwp 

:)

dg0yt reviewed (commented):
Mostly nitpick, given the amout of work.

But I would really appreciate to not change what doesn't need to be changed. Like extra lines in patches.

cenit committed:
Update ports/opencv/vcpkg.json

Co-authored-by: Kai Pastor <dg0yt@darc.de>

cenit committed:
Update ports/opencv/vcpkg.json

Co-authored-by: Kai Pastor <dg0yt@darc.de>

cenit committed:
Update ports/opencv/vcpkg.json

Co-authored-by: Kai Pastor <dg0yt@darc.de>

cenit committed:
Update ports/opencv2/usage.in

Co-authored-by: Kai Pastor <dg0yt@darc.de>

cenit committed:
Update ports/opencv4/usage.in

Co-authored-by: Kai Pastor <dg0yt@darc.de>

cenit committed:
Update ports/opencv4/vcpkg.json

Co-authored-by: Kai Pastor <dg0yt@darc.de>

cenit committed:
Update ports/opencv4/vcpkg.json

Co-authored-by: Kai Pastor <dg0yt@darc.de>

cenit committed:
Update ports/opencv4/vcpkg.json

Co-authored-by: Kai Pastor <dg0yt@darc.de>

cenit committed:
Update ports/opencv4/vcpkg.json

Co-authored-by: Kai Pastor <dg0yt@darc.de>

cenit committed:
Update ports/opencv3/usage.in

Co-authored-by: Kai Pastor <dg0yt@darc.de>

cenit commented on ports/opencv/portfile.cmake:
```diff
@@ -1,9 +1,8 @@
-if (NOT VCPKG_BUILD_TYPE)
-  file(INSTALL "${CURRENT_INSTALLED_DIR}/share/opencv4/OpenCVModules-debug.cmake" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}")
-endif()
+file(WRITE "${CURRENT_PACKAGES_DIR}/share/opencv/vcpkg-cmake-wrapper.cmake"
+[[
+set(OpenCV_DIR "${CMAKE_CURRENT_LIST_DIR}/../opencv4/" CACHE PATH "Path to OpenCVConfig.cmake" FORCE)
+set(OpenCV_ROOT "${CMAKE_CURRENT_LIST_DIR}/../opencv4/") #to force find_package(OpenCV) to look for specific version and not randomly pick one in between opencv* installed
+_find_package(${ARGS})
+]])
```
It's just a preference or a rule written anywhere? My preference goes to write them on the fly for such small snippets :)

dg0yt replied:
IMO it is a strong preference, also with an eye on future maintenance.

cenit committed:
fix references

cenit commented on ports/opencv/vcpkg.json:
```diff
@@ -292,6 +374,18 @@
         }
       ]
     },
+    "openjpeg": {
+      "description": "JPEG 2000 support for opencv",
```
no you're right. Jasper is possible only if OpenJPEG has not been found, otherwise OpenJPEG is preferred. What is the best way to encode this without restoring all the functions in the portfile that (correctly) Alexander told me to remove?

dg0yt replied:
There is no good way to encode alternatives in vcpkg. (I don't say I'm happy with it.)

cenit replied:
ok let's keep in this way for now, maybe in the future we just remove the jasper feature (now is marked as deprecated, i feel better to remove after one version has it marked so)

cenit committed:
Update ports/opencv4/vcpkg.json

Co-authored-by: Kai Pastor <dg0yt@darc.de>

cenit committed:
add missing support clause

cenit committed:
fix references

cenit committed:
fix python module path

cenit committed:
add another variable to fix python install path

cenit committed:
trying to fix errors related to absolutes path in python installed files

cenit committed:
fix references

cenit committed:
working on fixing absolute paths

cenit committed:
fix references

cenit committed:
fix errors in string path fixes

cenit committed:
fix references

cenit committed:
another try to fix python absolute paths

cenit committed:
Merge branch 'master' into dev/cenit/opencv49

cenit committed:
another try more

cenit committed:
fix references

cenit committed:
another round of fixes for python feature

cenit committed:
fix references

cenit commented:
@BillyONeal what's your opinion about the remaining open point? shall i write the file on the fly or shall i copy from the port folder?
checks are green, for me pr is complete otherwise

cenit committed:
remove deprecated features

cenit committed:
fix references

cenit committed:
do not write file on-the-fly

cenit committed:
fix references

cenit committed:
Merge branch 'master' into dev/cenit/opencv49

cenit committed:
bump port version

cenit committed:
fix references

cenit commented:
merged with master to fix conflicts

dg0yt commented:
Please note #41418

dg0yt commented:
In some patches, including port pcl, standard `git diff` header lines were removed from otherwise unchanged patches.
 - I really think this should not be applied at all to otherwise unchanged patches.
 - I think that it should not be applied if it requires manual editing every time we recreate a diff.

cenit commented:
> In some patches, including port pcl, standard `git diff` header lines were removed from otherwise unchanged patches.
> 
> * I really think this should not be applied at all to otherwise unchanged patches.
> * I think that it should not be applied if it requires manual editing every time we recreate a diff.

I just recreated the patch set of the ports I had to touch, I was not modifying anything in any untouched port otherwise.
My files are pure patch files, without any additional (unnecessary) data coming from git or other tools (whatever their reason might be for adding it).

cenit committed:
fix cmake IN list iterator

cenit committed:
fix references

cenit committed:
sorry i was just joking

cenit committed:
fix references

cenit committed:
Merge branch 'master' into dev/cenit/opencv49

theoractice commented:
Is this PR ready?

cenit commented:
> Is this PR ready?

for me yes

cenit committed:
Merge branch 'master' into dev/cenit/opencv49

cenit committed:
[oiio] bump port version
fix regex as in #41418

cenit commented:
merged with master to fix conflicts due to #41543

cenit commented:
errors popping up from osx pipelines seems unrelated to this PR

BillyONeal commented:
/azp run

BillyONeal commented:
Rerunning due to network mistake.

BillyONeal commented on ports/pcl/gh-6053.patch:
```diff
@@ -0,0 +1,17 @@
+     // Build camera filename
+     if (valid)
+     {
+-      unsigned int digest[5];
++      boost::uuids::detail::sha1::digest_type digest;
+       sha1.get_digest (digest);
+       sstream << ".";
+-      sstream << std::hex << digest[0] << digest[1] << digest[2] << digest[3] << digest[4];
++      for (int i = 0; i < 5; ++i) {
++        sstream << std::hex << *(reinterpret_cast<unsigned int*>(&digest[0]) + i);
++      }
```
This introduces undefined behavior which is unfortunate but it's what upstream accepted :(

BillyONeal reviewed (dismissed):
@JavierMatosD Do you remember why you tagged this team review?

russelltg commented:
I think the `string(APPEND DEPS_STRING "\nfind_dependency(CUDA)")` need to be `string(APPEND DEPS_STRING "\nfind_dependency(CUDAToolkit)")` now that we pass `ENABLE_CUDA_FIRST_CLASS_LANGUAGE`

cenit commented:
good catch, i think we can just remove the find dependency. Let me do some tests

cenit committed:
Merge branch 'master' into dev/cenit/opencv49

cenit committed:
remove unnecessary patch

cenit committed:
fix references

cenit commented:
merged with master to fix file conflicts (pcl patch was added also to a more recent PR already accepted)

cenit committed:
restore pcl port to master

cenit committed:
fix references

cenit committed:
apply review suggestion to find cudatoolkit dependency now that we use cuda first class language

cenit committed:
fix references

cenit commented:
> I think the `string(APPEND DEPS_STRING "\nfind_dependency(CUDA)")` need to be `string(APPEND DEPS_STRING "\nfind_dependency(CUDAToolkit)")` now that we pass `ENABLE_CUDA_FIRST_CLASS_LANGUAGE`

good catch. Done, thanks.

cenit committed:
remove unnecessary find_dependency call for CUDAToolkit

cenit committed:
fix references

cenit commented:
> > I think the `string(APPEND DEPS_STRING "\nfind_dependency(CUDA)")` need to be `string(APPEND DEPS_STRING "\nfind_dependency(CUDAToolkit)")` now that we pass `ENABLE_CUDA_FIRST_CLASS_LANGUAGE`
> 
> good catch. Done, thanks.

if CUDA feature is enable, this is already appearing in `OpenCVConfig.cmake`:

```
# Version Compute Capability from which OpenCV has been compiled is remembered
set(OpenCV_COMPUTE_CAPABILITIES "50-real;52-real;60-real;61-real;70-real;75-real;80-real;86-real;89-real;90-real;90-virtual")

set(OpenCV_CUDA_VERSION "12.6.20")
set(OpenCV_USE_CUBLAS   "1")
set(OpenCV_USE_CUFFT    "1")
set(OpenCV_USE_NVCUVID  "")
set(OpenCV_USE_NVCUVENC "")
set(OpenCV_CUDNN_VERSION    "")
set(OpenCV_USE_CUDNN        "")
set(ENABLE_CUDA_FIRST_CLASS_LANGUAGE  ON)

if(NOT CUDAToolkit_FOUND)
  if(NOT CMAKE_VERSION VERSION_LESS 3.18)
    if(UNIX AND NOT CMAKE_CUDA_COMPILER AND NOT CUDAToolkit_ROOT)
      message(STATUS "Checking for CUDAToolkit in default location (/usr/local/cuda)")
      set(CUDA_PATH "/usr/local/cuda" CACHE INTERNAL "")
      set(ENV{CUDA_PATH} ${CUDA_PATH})
    endif()
    find_package(CUDAToolkit ${OpenCV_CUDA_VERSION} EXACT REQUIRED)
  else()
    message(FATAL_ERROR "Using OpenCV compiled with CUDA as first class language requires CMake \>= 3.18.")
  endif()
else()
  if(CUDAToolkit_FOUND)
    set(CUDA_VERSION_STRING ${CUDAToolkit_VERSION})
  endif()
  if(NOT CUDA_VERSION_STRING VERSION_EQUAL OpenCV_CUDA_VERSION)
      message(FATAL_ERROR "OpenCV library was compiled with CUDA ${OpenCV_CUDA_VERSION} support. Please, use the same version or rebuild OpenCV with CUDA ${CUDA_VERSION_STRING}")
  endif()
endif()
```

so the `find_dependency(CUDA)` was unnecessary with the new architecture, but also the `find_dependency(CUDAToolkit)` is too!

I have removed it from OpenCV 4 port

cenit commented:
i think that regression on `vcpkg-ci-msys2` is not due to this PR

dg0yt commented:
> i think that regression on `vcpkg-ci-msys2` is not due to this PR

No, it isn't, and there are PRs for that.

dg0yt commented:
> > i think that regression on `vcpkg-ci-msys2` is not due to this PR
> 
> No, it isn't, and there are PRs for that.

Merged. (https://github.com/microsoft/vcpkg/pull/41691)

cenit committed:
Merge branch 'master' into dev/cenit/opencv49

cenit committed:
[darknet] bump version

cenit committed:
fix references

cenit committed:
fix error

cenit committed:
fix references

cenit committed:
fix arm64 regression, maybe

cenit committed:
fix references

cenit commented:
> @JavierMatosD Do you remember why you tagged this team review?

any news?

cenit commented:
@BillyONeal please let me know if you need any modification to this PR before considering it for merging.

cenit commented:
(after 4.9 there will be an immediate 4.10 opening...)

valorkin commented:
@cenit great work! much appreciated!

cenit commented:
@BillyONeal @vicroms @dan-shaw sorry for the periodic ping, but in case any modification is required I need to know it in advance to have some time for it. This PR is getting very late, due to me at the beginning being so slow in fixing regressions, but now it's stopped for no apparent reasons

OpenCV 4.10 is already out and OpenCV 4.11 is around the corner...

theoractice commented:
> @BillyONeal @vicroms @dan-shaw sorry for the periodic ping, but in case any modification is required I need to know it in advance to have some time for it. This PR is getting very late, due to me at the beginning being so slow in fixing regressions, but now it's stopped for no apparent reasons
> 
> OpenCV 4.10 is already out and OpenCV 4.11 is around the corner...

Maybe skip 4.10 and wait for 4.11 to save some time and effort. 
They are so slow.

Neumann-A commented on ports/opencv2/usage.in:
```diff
@@ -1,4 +1,4 @@
+@PORT@ is compatible with built-in CMake variables. In case of multiple OpenCV version installed, set OpenCV_ROOT *before* the find_package call:
 
-    set(OpenCV_DIR "${VCPKG_INSTALLED_DIR}/@TARGET_TRIPLET@/share/@PORT@")
+    set(OpenCV_ROOT "${VCPKG_INSTALLED_DIR}/@TARGET_TRIPLET@/share/@PORT@")
```
```suggestion
```


if vcpkg.cmake is used this should find the opencv from vcpkg otherwise VCPKG_INSTALLED_DIR is not set any way

cenit replied:
yes, of course the usage is referred to people using vcpkg.cmake
I assume we don't have to support people not using it, right?

Neumann-A commented on ports/opencv3/portfile.cmake:
```diff
@@ -39,97 +37,79 @@ else()
 endif()
 
-file(REMOVE "${SOURCE_PATH}/cmake/FindCUDNN.cmake")
+if (USE_QT_VERSION STREQUAL "6")
+  set(QT_CORE5COMPAT "Core5Compat")
+  set(QT_OPENGLWIDGETS "OpenGLWidgets")
+endif()
```
If it can use qt6, qt6 should be the default

cenit replied:
opencv3 can use qt6, but is not the preferred version (qt5 is preferred for the "old" opencv3)

Neumann-A commented on ports/opencv3/portfile.cmake:
```diff
@@ -39,97 +37,79 @@ else()
+ "openmp"          WITH_OPENMP
+ "ovis"            BUILD_opencv_ovis
+ "png"             WITH_PNG
+ "python"          BUILD_opencv_python3
+ "python"          WITH_PYTHON
```
This requires a dependency on python3[extensions]

Neumann-A commented on ports/opencv3/portfile.cmake:
```diff
@@ -208,34 +177,26 @@ if("contrib" IN_LIST FEATURES)
-        URLS "https://raw.githubusercontent.com/opencv/opencv_3rdparty/a56b6ac6f030c312b2dce17430eef13aed9af274/ippicv/ippicv_2020_mac_intel64_20191018_general.tgz"
-        FILENAME "opencv-cache/ippicv/1c3d675c2a2395d094d523024896e01b-ippicv_2020_mac_intel64_20191018_general.tgz"
-        SHA512 454dfaaa245e3a3b2f1ffb1aa8e27e280b03685009d66e147482b14e5796fdf2d332cac0f9b0822caedd5760fda4ee0ce2961889597456bbc18202f10bf727cd
+    vcpkg_download_distfile(OCV_DOWNLOAD
+      URLS "https://raw.githubusercontent.com/opencv/opencv_3rdparty/a56b6ac6f030c312b2dce17430eef13aed9af274/ippicv/ippicv_2020_mac_intel64_20191018_general.tgz"
+      FILENAME "opencv-cache/ippicv/1c3d675c2a2395d094d523024896e01b-ippicv_2020_mac_intel64_20191018_general.tgz"
+      SHA512 454dfaaa245e3a3b2f1ffb1aa8e27e280b03685009d66e147482b14e5796fdf2d332cac0f9b0822caedd5760fda4ee0ce2961889597456bbc18202f10bf727cd
```
ipp should probably its own port in the future or does opencv add something ontop?

cenit replied:
ipp-icv is a subset of ipp dedicated to computer vision. 

Per intel pages (https://www.intel.com/content/www/us/en/developer/articles/troubleshooting/intel-integrated-performance-primitives-intel-ipp-open-source-computer-vision-library-opencv-faq.html):
"The IPP-ICV package is provided with the OpenCV library,... , and users do not need install additional standalone Intel® IPP package."

There is no ipp-icv package outside of opencv, and I am really unsure a generic ipp can integrate properly into each opencv revision (ipp-icv is tight to opencv per revision history, intel is heavily involved in opencv)

Neumann-A commented on ports/opencv3/portfile.cmake:
```diff
@@ -299,24 +228,19 @@ if("qt" IN_LIST FEATURES)
         -DX86_64=${TARGET_IS_X86_64}
         -DX86=${TARGET_IS_X86}
         -DARM=${TARGET_IS_ARM}
+        ###### use c++17 to enable features that fail with c++11 (halide, protobuf, etc.)
+        -DCMAKE_CXX_STANDARD=17
```
I feel like those ports need to communicate this requirement in the future

cenit replied:
you're right. I don't like to force cpp17 downstream simply because opencv does not need it. It's its dependencies that have now adopted cpp17 and require downstream projects to enable it to let their header work when embedded, and so a transitive communication would be perfect

Neumann-A commented on ports/opencv3/portfile.cmake:
```diff
@@ -335,74 +259,74 @@ vcpkg_cmake_configure(
         -DCMAKE_DISABLE_FIND_PACKAGE_JNI=ON
         # ENABLE
-        -DENABLE_CXX11=${ENABLE_CXX11}
+        -DENABLE_CXX11=ON
```
doesnt this conflict with         
```
###### use c++17 to enable features that fail with c++11 (halide, protobuf, etc.)
        -DCMAKE_CXX_STANDARD=17
```
above?

cenit replied:
seems not.

Neumann-A commented on ports/opencv3/portfile.cmake:
```diff
@@ -438,105 +361,210 @@ if(${BUILD_opencv_flann} AND NOT TARGET libprotobuf) #Check if the CMake target
+if("omp" IN_LIST FEATURES)
+  string(APPEND DEPS_STRING "\nfind_dependency(OpenMP)")
+endif()
+if("ovis" IN_LIST FEATURES)
+  string(APPEND DEPS_STRING "\nfind_dependency(OGRE)")
```
ovis == ogre?

Neumann-A commented on ports/opencv3/usage.in:
```diff
@@ -1,4 +1,4 @@
+@PORT@ is compatible with built-in CMake variables. In case of multiple OpenCV version installed, set OpenCV_ROOT *before* the find_package call:
 
-    set(OpenCV_DIR "${VCPKG_INSTALLED_DIR}/@TARGET_TRIPLET@/share/@PORT@")
+    set(OpenCV_ROOT "${VCPKG_INSTALLED_DIR}/@TARGET_TRIPLET@/share/@PORT@")
```
> In case of multiple OpenCV version installed, set OpenCV_ROOT *before* the find_package call

this should probably be handled by the VERSION number in find_package somehow

cenit replied:
it's not.

Neumann-A commented on ports/opencv3/vcpkg.json:
```diff
@@ -205,10 +187,12 @@
       "description": "opencv_ovis module",
+      "supports": "!(windows & static) & !android",
       "dependencies": [
         "ogre",
```
ok here is ogre== ovis

cenit replied:
https://docs.opencv.org/4.x/d2/d17/group__ovis.html

"ovis is a simplified rendering wrapper around [ogre3d](https://www.ogre3d.org/). The [Ogre terminology](https://ogrecave.github.io/ogre/api/latest/_the-_core-_objects.html) is used in the API and [Ogre Script](https://ogrecave.github.io/ogre/api/latest/_scripts.html) is assumed to be used for advanced customization"

Neumann-A commented on ports/opencv3/vcpkg.json:
```diff
@@ -228,6 +212,7 @@
     },
     "python": {
```
depend on python3[extensions]

Neumann-A commented on ports/opencv3/vcpkg.json:
```diff
@@ -228,6 +212,7 @@
     },
     "python": {
       "description": "Python wrapper support for opencv",
+      "supports": "!(windows & static)",
```
depend on python3[extensions] instead

Neumann-A commented on ports/opencv4/vcpkg.json:
```diff
@@ -300,6 +361,7 @@
     },
     "python": {
       "description": "Python wrapper support for opencv",
+      "supports": "!(windows & static)",
```
same as above, dependency on python3[extensions] instead of supports

cenit replied:
ok did it

JavierMatosD commented:
> any news?

We are experimenting with a new internal review process. We were using `vcpkg-team-review` to reflect something else. 

This PR lgtm, but I'd prefer that @Neumann-A's review comments be addressed. I'm placing the PR in draft for now. **Please mark 'ready for review' when you have either addressed Neumann-A's feedback or responded. I will not look at this PR if it is in Draft. Marking it 'ready for review' lets me know that it is ready to be looked at again. **

JavierMatosD reviewed (dismissed):
Just need a response to Neumann-A's review

cenit committed:
Merge branch 'master' into dev/cenit/opencv49

cenit committed:
use python3[extensions] instead of python3 and support !windows&static clause

cenit committed:
fix references

cenit commented:
I tried to answer all points and I applied a modification for what concerns the python3[extensions] clause.
Let's wait for CI, then if @Neumann-A is fine I will make the PR ready for review again

cenit commented:
assuming the approval as a generic go for all open points, i resolved all of them and marked pr as ready

AdrianEddy commented:
@cenit I'd like to express my gratitude for you for putting in all the work on this PR and sticking with it for all the needed changes and fixes. I was watching it regularly and seeing you going through so many reference updates, issues with dependencies and extremely long CI runs and I just wanted to say thank you. You did a great job for all of us!

theoractice commented:
@cenit thank you for your contribution!
However I just found this update installs all opencv headers into `opencv4` directory on windows. Needs a fix?
I am using the latest vcpkg.exe.

dg0yt commented:
I'm happy to see opencv 4.9 merged. However:

 - I see downstream regressions in some PRs which involve opencv4 at least indirectly due to changes to ci baseline, e.g. #41525.
 - I have a direct fail for opencv4[core,tesseract]:x64-android (during analysis of the other failures).
 - I stumble again over the updated patches with no relevant changes. They really make it harder to analyse what has changed and what hasn't.

cenit commented:
> I'm happy to see opencv 4.9 merged. However:
> 
> * I see downstream regressions in some PRs which involve opencv4 at least indirectly due to changes to ci baseline, e.g. [[itk] Revise, test, fix #41525](https://github.com/microsoft/vcpkg/pull/41525).
> * I have a direct fail for opencv4[core,tesseract]:x64-android (during analysis of the other failures).
> * I stumble again over the updated patches with no relevant changes. They really make it harder to analyse what has changed and what hasn't.

openvino is a dependency of opencv, so if opencv can be a dependency of openvino (i don't know it very well) that might create a circular dependency that vcpkg has many times failed to serve properly (it would require building base opencv, then openvino, then again opencv with openvino). Creating openvino forcing it without opencv might be the best solution if we cannot solve the circular problem above.

opencv4 has no tesseract feature. Which fail do you notice? I tried many combinations and all were working. I found that with opencv 4.10 I had to change something in opencv4[core,contrib], which has a dependency on tesseract. Is it what you meant? I didn't find this problem with OpenCV 4.9 when i tested it in the same test matrix i run...

which patches have no relevant changes here? i will try my best to help you find any regression that might have been lost, even before the #41985 gets interesting

dg0yt commented:
> Creating openvino forcing it without opencv might be the best solution if we cannot solve the circular problem above.

It is a requirement. Implemented in #42017.

> opencv4 has no tesseract feature. ... opencv4[core,contrib], which has a dependency on tesseract.

Yes. https://github.com/microsoft/vcpkg/pull/42015.

> which patches have no relevant changes here?

To clarify: no relevant changes wrt to the patch for 4.8. Such as [ports/opencv4/0006-fix-uwp.patch](https://github.com/microsoft/vcpkg/pull/38928/files#diff-624dbdb14e5323dbfa870b557bf33034672fd5c0d8e8574d626f3ee3320635ff).

dg0yt commented:
FTR I really don't expect every port and feature configuration to work immediately after a major update, given the complexity fo the ports.

cenit commented:
About #42015, strange that i didn't notice it in the test matrix and it appeared for the 4.10... sorry for that. It's not a config set we use very much, thanks for the PR

About the relevant change...

![image](https://github.com/user-attachments/assets/510e1b91-36ea-471d-915f-75a10754700c)

this is the relevant one: the new line to which apply the fix. Some configs are very strict about patch application, if the line does not match exactly they fail, and I prefer to be compliant with those requirements.

dg0yt commented:
FTR the openvino failure is quite interesting:
~~~
CMake Error at /vcpkg/downloads/tools/cmake-3.30.1-linux/cmake-3.30.1-linux-x86_64/share/cmake-3.30/Modules/FindOpenSSL.cmake:210 (pkg_check_modules):
  Unknown CMake command "pkg_check_modules".
Call Stack (most recent call first):
  /mnt/vcpkg-ci/installed/x64-android/share/openssl/vcpkg-cmake-wrapper.cmake:37 (_find_package)
  /vcpkg/scripts/buildsystems/vcpkg.cmake:813 (include)
  /vcpkg/downloads/tools/cmake-3.30.1-linux/cmake-3.30.1-linux-x86_64/share/cmake-3.30/Modules/CMakeFindDependencyMacro.cmake:76 (find_package)
  /mnt/vcpkg-ci/installed/x64-android/share/curl/CURLConfig.cmake:60 (find_dependency)
  /mnt/vcpkg-ci/installed/x64-android/share/curl/vcpkg-cmake-wrapper.cmake:11 (_find_package)
  /vcpkg/scripts/buildsystems/vcpkg.cmake:813 (include)
  /vcpkg/downloads/tools/cmake-3.30.1-linux/cmake-3.30.1-linux-x86_64/share/cmake-3.30/Modules/CMakeFindDependencyMacro.cmake:76 (find_package)
  /mnt/vcpkg-ci/installed/x64-android/share/tesseract/TesseractConfig.cmake:24 (find_dependency)
  /vcpkg/scripts/buildsystems/vcpkg.cmake:859 (_find_package)
  /vcpkg/downloads/tools/cmake-3.30.1-linux/cmake-3.30.1-linux-x86_64/share/cmake-3.30/Modules/CMakeFindDependencyMacro.cmake:76 (find_package)
  /mnt/vcpkg-ci/installed/x64-android/share/opencv4/OpenCVModules.cmake:41 (find_dependency)
  /mnt/vcpkg-ci/installed/x64-android/share/opencv4/OpenCVConfig.cmake:139 (include)
  /mnt/vcpkg-ci/installed/x64-android/share/opencv/vcpkg-cmake-wrapper.cmake:3 (_find_package)
  /vcpkg/scripts/buildsystems/vcpkg.cmake:813 (include)
  cmake/extra_modules.cmake:7 (find_package)
  cmake/extra_modules.cmake:177 (ov_generate_dev_package_config)
  CMakeLists.txt:165 (include)
~~~
The only scenario where this should happen is with `CMAKE_DISABLE_FIND_PACKAGE_PkgConfig=ON`,
I didn't want to debug this any further. Just quickly fixing the baseline regression.

cenit commented:
i was constantly getting this regression with openvino on android in this pr if you remember. I wrote also about it along the thread. Then it disappeared

dg0yt commented:
Hm, I find the old posts now, but I have no memories. That's remarkable, given https://gitlab.kitware.com/cmake/cmake/-/merge_requests/9709. But that was triggered from changes in CURL.

So with the next CMake version in vcpkg, this particular error will no longer occur...

sandye51 commented:
> > I'm happy to see opencv 4.9 merged. However:
> > 
> > * I see downstream regressions in some PRs which involve opencv4 at least indirectly due to changes to ci baseline, e.g. [[itk] Revise, test, fix #41525](https://github.com/microsoft/vcpkg/pull/41525).
> > * I have a direct fail for opencv4[core,tesseract]:x64-android (during analysis of the other failures).
> > * I stumble again over the updated patches with no relevant changes. They really make it harder to analyse what has changed and what hasn't.
> 
> openvino is a dependency of opencv, so if opencv can be a dependency of openvino (i don't know it very well) that might create a circular dependency that vcpkg has many times failed to serve properly (it would require building base opencv, then openvino, then again opencv with openvino). Creating openvino forcing it without opencv might be the best solution if we cannot solve the circular problem above.
> 
> opencv4 has no tesseract feature. Which fail do you notice? I tried many combinations and all were working. I found that with opencv 4.10 I had to change something in opencv4[core,contrib], which has a dependency on tesseract. Is it what you meant? I didn't find this problem with OpenCV 4.9 when i tested it in the same test matrix i run...
> 
> which patches have no relevant changes here? i will try my best to help you find any regression that might have been lost, even before the #41985 gets interesting

OpenCV is used only in testing / sampler by OpenVINO, so it's not a produce level dependency. 
As tests are turned off in vcpkg is the right decision.

moratom commented on ports/rtabmap/vcpkg.json:
I'm new with vcpkg, so apologies in advance if this is a stupid question.

I stumbled upon this PR when updating the baseline to latest, since `RTABMAP` doesn't build anymore.
Shouldn't we now we have more features specified here, since many were moved out of "core"?
